src/cpp/desktop/ScriptCommand.mm

```diff
@@ -0,0 +1,55 @@
+/*
+ * ScriptCommand.mm
+ *
+ * Copyright (C) 2009-18 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+#import "ScriptCommand.h"
+#include "DesktopMainWindow.hpp"
+#include <boost/algorithm/string/replace.hpp>
+
+
+std::string jsEscape(std::string str)
```

Comment:
We have a couple escape helpers here: do any of them suffice?

https://github.com/rstudio/rstudio/blob/f5ba8af725c845a1097277ad3f812103b8ab090f/src/cpp/core/include/core/StringUtils.hpp#L91-L95